### DISCUSSION TITLE
cdh - remember history

### DISCUSSION THREAD

**Discussion by backspacedodge** [fish-shell/fish-shell#10765] [Q&A] [answered]
I like the idea portrayed in https://fishshell.com/docs/current/cmds/cdh.html to have access to recently visited directories in addition to the fantastic completion that fish provides. I noticed that every time I close and reopen my terminal, the history of directories I've visited is flushed and I start from zero.
```
The history is stored in the dirprev and dirnext variables, which this command manipulates. If you make those universal variables, your cd history is shared among all fish instances.
```
Is the above section implying that creating universal variables would allow me to have a "history", meaning -- I can close all running terminals and when I open a new one, it'll remember previous directories?

To that end, I added the following to `~/.config/fish/config.fish` and I still seem to start from zero when all terminals are closed and then start with a new one. Does anyone know what I could be missing?

```
set -Ux dirprev
set -Ux dirnext
```

**Comment by lpar** [comment, marked as answer]
I just solved this one. In `config.fish` you want to do:

```
set -Ux dirprev $dirprev
set -Ux dirnext $dirnext
```

That should keep existing values while ensuring the variables are universal.
